### PR TITLE
SQL: [Tests] Fix and enable internalClusterTests

### PR DESCRIPTION
**Pull request by matriv**
Fixes: #37191

### CODE REVIEW

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/LocalStateSQLXpackPlugin.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+package org.elasticsearch.xpack.sql.action;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xpack.core.LocalStateCompositeXPackPlugin;
+import org.elasticsearch.xpack.sql.plugin.SqlLicenseChecker;
+import org.elasticsearch.xpack.sql.plugin.SqlPlugin;
+
+import java.nio.file.Path;
+
+public class LocalStateSQLXpackPlugin extends LocalStateCompositeXPackPlugin {
```
Comment (astefan) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/LocalStateSQLXpackPlugin.java]: Does it have to be `public`? Also, I think the naming around xpack should be XPack (capital `P`).

**Review by astefan — APPROVED** [review]
LGTM. Left one comment.

**Comment by matriv** [comment]
run default distro tests

**Comment by matriv** [comment]
run gradle build tests 2

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/LocalStateSQLXpackPlugin.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/LocalStateSQLXpackPlugin.java]: no, can be package private, thx!

**Comment by matriv** [comment]
run default distro tests

**Comment by matriv** [comment]
run gradle build tests 2

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/LocalStateSQLXpackPlugin.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/LocalStateSQLXpackPlugin.java]: Actually, it needs to be public:
```
   > Throwable #1: java.lang.IllegalStateException: no public constructor for [org.elasticsearch.xpack.sql.action.LocalStateSQLXPackPlugin]
   >    at org.elasticsearch.plugins.PluginsService.loadPlugin(PluginsService.java:590)
   >    at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:119)
   >    at org.elasticsearch.node.Node.<init>(Node.java:303)
```

**Comment by matriv** [comment]
Seems that the changes affect the `integTestCluster`, so Marked it as WIP for the moment.

**Comment by matriv** [comment]
With the awesome help of @hub-cap the issue has been fixed.
The PR is ready for review.

Also opened a new issue: https://github.com/elastic/elasticsearch/issues/37320 to fix the licencing tests.

**Comment by matriv** [comment]
@astefan Maybe you want to also take another look here.

**Review by costin — APPROVED** [review]
LGTM.

**Comment by matriv** [comment]
run gradle build tests 2

**Comment by matriv** [comment]
run gradle build tests 1

**Comment by astefan** [comment]
@matriv still LGTM

**Comment by matriv** [comment]
Backported to `6.x` with 5e17a7846589a7028d94f708128866863a6a81ec

**Pull request merged**